ensime_shared/client.py

```diff
@@ -249,7 +249,7 @@ def send_at_position(self, what, useSelection, where="range"):
         self.log.debug('send_at_position: in')
         b, e = self.editor.selection_pos() if useSelection else self.editor.word_under_cursor_pos()
         self.log.debug('useSelection: {}, beg: {}, end: {}'.format(useSelection, b, e))
-        self.send_at_point_req(what, self.editor.path(), b[0], b[1], e[0], e[1], where)
+        self.send_at_range_req(what, self.editor.path(), b[0], b[1], e[0], e[1], where)
```

Comment:
Ok, so the thing is that at_range was only used once within send_at_position so I got rid of it and inlined the code in the later which is indeed called with 'point' a few times and in one case it uses the default value:

```
ensime_shared/client.py|357 col 14| self.send_at_position("Type", useSelection)
ensime_shared/client.py|436 col 14| self.send_at_position("DocUri", False, "point")
ensime_shared/client.py|451 col 14| self.send_at_position("DocUri", False, "point")
```

I've added some docs to the method to clarify the use of the parameters, should be a bit clearer now, still not ideal and I would like to improve it but any further refactor is out of the scope of this PR IMHO.

So now we only have two separate send_at... methods and they have clear different responsibilities, one sends `from` and `to` based on selection or word under cursor some times calling the field 'range' and other 'point, the other method simply sends the cursor position coordinates.